sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverClientIntegrationTest.java

```diff
@@ -0,0 +1,578 @@
+        // Arrange
+        setSenderAndReceiver(entityType, isSessionEnabled);
+        int howManyMessage = 2;
+        int noMessages = 0;
+
+        // Act
+        IterableStream<ServiceBusReceivedMessage> iterableMessages = receiver.receive(howManyMessage, Duration.ofSeconds(15));
+
+        // Assert
+        final int receivedMessages = iterableMessages.stream().collect(Collectors.toList()).size();
+        assertEquals(noMessages, receivedMessages);
+    }
+
+    /**
+     * Verifies that we can send and receive one messages.
+     */
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void receiveMessageAutoComplete(MessagingEntityType entityType, boolean isSessionEnabled) {
+        // Arrange
+        setSenderAndReceiver(entityType, isSessionEnabled);
+        int howManyMessage = 1;
+
+        final String messageId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = getMessage(messageId, isSessionEnabled);
+
+        sendMessage(message);
+
+        // Act
+        Iterable<ServiceBusReceivedMessage> iterableMessages = receiver.receive(howManyMessage, TIMEOUT);
+
+        // Assert
+        for (ServiceBusReceivedMessage receivedMessage: iterableMessages) {
+            assertMessageEquals(receivedMessage, messageId, isSessionEnabled);
+        }
+
+        messagesPending.decrementAndGet();
+        messagesPending.decrementAndGet();
+    }
+
+    /**
+     * Verifies that we can send and peek a message.
+     */
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void peekMessage(MessagingEntityType entityType, boolean isSessionEnabled) {
+        // Arrange
+        setSenderAndReceiver(entityType, isSessionEnabled);
+
+        final String messageId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = getMessage(messageId, isSessionEnabled);
+
+        sendMessage(message);
+
+        // Act
+        ServiceBusReceivedMessage receivedMessage = receiver.peek();
+
+        // Assert
+        assertMessageEquals(receivedMessage, messageId, isSessionEnabled);
+    }
+
+    /**
+     * Verifies that we can send and peek a message.
+     */
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void peekFromSequenceNumberMessage(MessagingEntityType entityType, boolean isSessionEnabled) {
+        // Arrange
+        setSenderAndReceiver(entityType, isSessionEnabled);
+
+        final String messageId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = getMessage(messageId, isSessionEnabled);
+
+        sendMessage(message);
+
+        final IterableStream<ServiceBusReceivedMessage> messageIte = receiver.receive(1, TIMEOUT);
+        Assertions.assertNotNull(messageIte);
+
+        final List<ServiceBusReceivedMessage> asList = messageIte.stream().collect(Collectors.toList());
+        ServiceBusReceivedMessage receivedMessage = asList.get(0);
+
+        assertNotNull(receivedMessage);
```

Comment:
Is this assertion necessary? If the list was empty, the .get(0) would fail with an out of range exception. The assertion is probably `Assertions.assertEquals(1, asList.size())` before you get the first item.